compliance_checker/cf/cf.py

```diff
@@ -3099,14 +3099,20 @@ def check_cf_role(self, ds):
         """
         ret_val = []
         valid_roles = ['timeseries_id', 'profile_id', 'trajectory_id']
+        variable_count = 0
         for variable in ds.get_variables_by_attributes(cf_role=lambda x: x is not None):
+            variable_count += 1
             name = variable.name
             valid_cf_role = TestCtx(BaseCheck.HIGH, '§9.5 {} contains a valid cf_role attribute'.format(name))
             cf_role = variable.cf_role
             valid_cf_role.assert_true(cf_role in valid_roles,
                                       "{} is not a valid cf_role value. It must be one of {}"
                                       "".format(name, ', '.join(valid_roles)))
-        return ret_val
+        valid_cf_role.assert_true(variable_count < 2,
+                                  ('§9.5 states that datasets should not '
+                                   'contain more than two variables defining a '
+                                   'cf_role attribute.'))
+        return valid_cf_role.to_result()
```

Comment:
@lukecampbell Do we need to know the featureType to check the correct number of occurrences of cf_role (1 or 2)? 
```
CF files that contain timeSeries, profile or trajectory featureTypes, should include only a single occurrence of a cf_role attribute; CF files that contain timeSeriesProfile or trajectoryProfile may contain two occurrences
```